### PR TITLE
Add information about the convenience API to the docs

### PR DESCRIPTION
**Pull request by jdfreder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
...
+  console.log('is child');
+}
+```
+
+It will probably make more sense to use it as an observable filter.  In the example below, `parent` is a [Jupyter message object](https://ipython.org/ipython-doc/3/development/messaging.html#general-message-format) and `channels.iopub` is an RxJS observable:
+
+```js
+const enchannel = require('enchannel');
+const isChildMessage = enchannel.isChildMessage.bind(null, parent);
+const childMessages = channels.iopub.filter(isChildMessage);
```
Comment (rgbkrk) [inline README.md:96]: We'll probably want to show how to use this as an extension on Observables and ES7 bind, though I suppose I still haven't figured those all the way out (as a best practice).

File: README.md [hunk 2]
```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
...
+
+ - username, string  
+ - session, string,  guid unique to the current session  
+ - msg_type: string, type of the message getting sent  
+
+The following is a full example that shows how connection may happen, how you'd setup the session and username, and then create and send a shutdown request:
+
+```js
+// The method to connect is specific to the enchannel backend implementation
+enchannelBackend.connect().then(channels => {
```
Comment (rgbkrk) [inline README.md:110]: Can we rely on channels being setup already for `createMessage`, to keep the code snippet focused on what the reader is supposed to get out of this section?

Comment (jdfreder) [inline README.md:110]: :+1:

File: README.md [hunk 3]
```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
...
+
+  // Created once with the channels
+  const uuid = require('node-uuid');
+  const session = uuid.v4();
+  const username = process.env.LOGNAME || process.env.USER ||
+    process.env.LNAME || process.env.USERNAME;
+
+  // Create the shutdown request method
+  const enchannel = require('enchannel');
+  const shutdownRequest = enchannel.createMessage(username, session, 'shutdown_request');
```
Comment (rgbkrk) [inline README.md:120]: The way I'm going to be tempted to use this in the future is to bind:

``` js
const createMessage = enchannel.createMessage.bind(null, username, session)
```

Which means that I :heart: the order here. At runtime I don't expect the username to change, though the session will.

File: README.md [hunk 4]
```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
...
+ - username, string  
+ - session, string,  guid unique to the current session  
+ - channels: object, enchannel channels object
+ - restart: optional boolean, whether the shutdown request is actually a restart request
+
+The following full example shows how this method would be used:
+
+```js
+// The method to connect is specific to the enchannel backend implementation
+enchannelBackend.connect().then(channels => {
```
Comment (rgbkrk) [inline README.md:144]: Same "give me the `channels`" bit here too.

Maybe start with:

``` js
channels = ...connected using an enchannel backend...
```

Comment (rgbkrk) [inline README.md:144]: I recognize this is almost always going to be wrapped in a `Promise`, that part can come up in the `enchannel-*-backend`s.

File: README.md [hunk 5]
```diff
@@ -70,6 +70,108 @@ Messages observed from these Subjects are all immutable, not by convention but t
...
+## Development
+
+To contribute to the spec or utility functions, clone this repo and install it by running the following from the repo root:
+
+    npm install
+
+Before contributing any changes to the utility functions, make sure the unit tests pass locally by running:
+
+    npm test
+
```
Comment (rgbkrk) [inline README.md:174]: :+1:

File: README.md (hunk 4 above)
Comment (jdfreder) [inline README.md:144]: :+1:

**Pull request merged**